### ISSUE TITLE
Add Firefox Android Team Tag (FXDROID)

### ISSUE THREAD

**Issue by cpeterso** [mozilla/jira-bugzilla-integration#301]
The Firefox Android team would like to set up JBI sync from Bugzilla to our [FXDROID Jira project](https://mozilla-hub.atlassian.net/plugins/servlet/project-config/FXDROID/summary). The Android team will typically work in the `Fenix`, `Focus`, and `GeckoView` Bugzilla products, but it would be nice if we could also sync Bugzilla bugs in other products (such as `Core`).

Something like this untested code:

```
whiteboard_tag: fxdroid
  contact: cpeterson@mozilla.com
  description: Firefox Android Team Tag
  enabled: true
  parameters:
    jira_project_key: FXDROID
   steps:
      new:
      - create_issue
      - maybe_delete_duplicate
      - add_link_to_bugzilla
      - add_link_to_jira
      - maybe_assign_jira_user
      - maybe_update_issue_resolution
      - maybe_update_issue_status
      existing:
      - update_issue
      - maybe_assign_jira_user
      - maybe_update_issue_resolution
      - maybe_update_issue_status
    status_map:
      UNCONFIRMED: Backlog
      NEW: Backlog
      ASSIGNED: In Progress
      RESOLVED: Closed
      REOPENED: In Progress
      VERIFIED: Closed
    resolution_map:
      FIXED: Fixed
      INVALID: Won't Do
      WONTFIX: Won't Do
      INACTIVE: Won't Do
      DUPLICATE: Duplicate
      WORKSFORME: Cannot Reproduce
      INCOMPLETE: Won't Do
      MOVED: Duplicate
```

1. I think I may be trying to use Jira status and resolution incorrectly. IIUC, the [FXDROID Jira project](https://mozilla-hub.atlassian.net/plugins/servlet/project-config/FXDROID/summary) has separate `Done` and `Cancelled` status values, but maybe `Cancelled` is redundant and we should use the `Done` status with a resolution like "Won't Do" and "Duplicate"?

2. What are the default `steps` for `new`, `existing`, and `comment`? Do nothing?

3. Is it possible to set the Jira issue type based on the Bugzilla bug type (`defect`, `enhancement`, `task`)?

![image](https://user-images.githubusercontent.com/629193/198152567-c5ef0f02-51e5-4c29-8889-eef91aab6a24.png)

If it’s not possible, that’s OK; it's not a big deal. If it is possible, I’d like to map Bugzilla bug types to Jira issue types like this:

```
  defect: Bug
  enhancement: Task
  task: Task
```

**Comment by bsieber-mozilla** [comment]
1. I've [requested](https://mozilla-hub.atlassian.net/servicedesk/customer/portal/4/SDD-9576?created=true) the FXDROID project be duplicated, with the current configuration style, to the testing environment; that way the workflows/transitions can be tested and confirmed. 
2. The defaults are listed below; they can also be found in the code [here](https://github.com/mozilla/jira-bugzilla-integration/blob/main/jbi/actions/default.py#L28)
    - _Steps below refer to the python functions in [this module](https://github.com/mozilla/jira-bugzilla-integration/blob/main/jbi/actions/steps.py )_
    - "new" will use the following steps, by default:
      - create_issue
      - maybe_delete_duplicate
      - add_link_to_bugzilla
      - add_link_to_jira
    - "existing" will use the following steps, by default:
      - update_issue
      - add_jira_comments_for_changes
    - "comment" will use the following step, by default:
      - create_comment
3. Currently it is not possible to set the Jira issue type based on the Bugzilla bug type; however, the map you provided aligns with the defaulted behavior; the map takes place in [this method](https://github.com/mozilla/jira-bugzilla-integration/blob/d8295054cbaf31155c5b3576cbfac05f8d61e9aa/jbi/models.py#L280-L283).

**Comment by cpeterso** [comment]
@bsieber-mozilla Thanks. I updated the yaml in the first comment to remove the incorrect usage of `resolution_map`. I think the yaml now correctly describes the config I'd like for FXDROID syncing.

**Comment by bsieber-mozilla** [comment]
@cpeterso that yaml is looking nearly perfect! 

There's one thing that we may have to change, which is the `whiteboard-tag` 

"-" are a special character used often in whiteboard-tags, however original usage intended the whiteboard-tag be of the form: [TAG-metadata]; where "-metadata" was used by PM's and other's but not the jbi system. 

So JBI would look at any precursor to a "-" between bracketed as the routing/running component mapping to an action.

---

tldr;

currently "-" is reserved--explained more above; 
do you have a preference between `fxdroid_jira`,`fxdroid`, or perhaps a different tag?

**Comment by cpeterso** [comment]
> do you have a preference between `fxdroid_jira`,`fxdroid`, or perhaps a different tag?

I see. If we use `fxdroid`, does that mean`fxdroid` will match the substring within `FOOfxdroidBAR`? Or the bare word `fxdroid` without square brackets in `foo fxdroid bar`?

I was emulating the Add-ons team's use of `[addons-jira]` in the Bugzilla whiteboard, but I see now that their JBI tag is actually just `addons` and `jira` is the metadata:

https://github.com/mozilla/jira-bugzilla-integration/blob/04240cf993b028dcf85438b9446154ad14244e61/config/config.prod.yaml#L3

**Comment by cpeterso** [comment]
> do you have a preference between fxdroid_jira,fxdroid, or perhaps a different tag?

@bsieber-mozilla: Let's use `fxdroid`. I updated the yaml in my first comment to use `whiteboard_tag: fxdroid`.

**Comment by cpeterso** [comment]
I also added a `resolution_map` to the first comment that maps Bugzilla resolutions to similar Jira resolutions.

**Comment by cpeterso** [comment]
@bsieber-mozilla I see PR #307 has been merged. How long until the new JBI config is live in production?

**Comment by bsieber-mozilla** [comment]
Hi @cpeterso, apologies for the delay, I've been on medical leave/ooo since Oct 30. I hope to be back at my desk later this week.

I've tried to push this forward since, but reaching out to @grahamalama or @leplatrem might expedite this.

In terms of your question; as long as you've tested the tag and it is as desired (in the bugzilla-dev and mozit-test Jira env) this will just require a git release.

**Comment by cpeterso** [comment]
> In terms of your question; as long as you've tested the tag and it is as desired (in the bugzilla-dev and mozit-test Jira env) this will just require a git release.

@bsieber-mozilla: I don't know how to test bugzilla-dev and mozit-test Jira env. Do I need to do that before you can release this JBI change?

**Comment by bsieber-mozilla** [comment]
Hi @cpeterso, I attempted to test this workflow in non-prod and it failed due to limitations with the current permission scopes of the API key; we're working with Jira Support to ensure the correct permissions are applied to all environments.

Since our prior internal Jira Support agent is no longer with the company, new Jira Support have stepped in to assist. The permissions seem to be missing in the production instance as well (given the state of the heartbeat and [powered_by_jbi](https://jbi.services.mozilla.com/powered_by_jbi/) pages.

**Comment by bsieber-mozilla** [comment]
Hi @cpeterso, Jira Support has updated the permissions in both the nonprod and prod environments.

I began to test some of the [FXDROID steps in nonprod](https://bugzilla-dev.allizom.org/show_bug.cgi?id=1632536)--but ran into an issue with the setup; it seems that `resolutions` were not set for the project (in non-prod) so I am unable to test the resolution map successfully. (As to why the "Done" status has yet to be sent in the issue test: the act of "Resolving" a bug sends a status and resolution, the bot attempts to set the `resolution` field before the `status` field and results in a failure, halting the workflow. )

> You may notice some duplicates on the [nonprod jira board](https://mozit-test.atlassian.net/jira/software/c/projects/FXDROID/boards/515?config=detailView&insightsConfigTab=board&selectedIssue=FXDROID-11) that occurred when permissions were missing while tests were attempted at an earlier time. 


One approach is: we remove the resolution_map, confirm this is working as expected, and then users can begin to use the [fxdroid] tag--without resolutions.

Another approach might be: work with Jira Support to add the expected `resolution` field to the Jira Project.

_I believe the first approach can be done and applied to both environments to move this forward and allow for some testing and usage, while the second approach can be a future addition._

----------
We make the call to update resolutions here:
https://github.com/mozilla/jira-bugzilla-integration/blob/f7194e05bd9ec11eeee992ed08027424628321cb/jbi/services/jira.py#L398

through the atlassian api client:
https://github.com/atlassian-api/atlassian-python-api/blob/0dfb78a99980f2d6a7e02ac5166197d99cfea359/atlassian/jira.py#L1079

which uses the jira api:
https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issues/#api-rest-api-2-issue-issueidorkey-put

**Comment by bsieber-mozilla** [comment]
(Also, `Focus`, `Fenix`, `GeckoView` and `Core` bugzilla product's webhooks are setup in both nonprod and prod to broadcast events to JBI.)

**Comment by bsieber-mozilla** [comment]
I opened this PR removing the resolution_map for the time being; we can merge this in if you're comfortable with the first approach.

**Comment by cpeterso** [comment]
> I opened this PR (#319 ) removing the resolution_map for the time being; we can merge this in if you're comfortable with the first approach.

That sounds OK to me. The status_map is more important than the resolution_map, so let's not block on the resolution_map.

**Comment by bsieber-mozilla** [comment]
With the most recent release, yesterday afternoon, this workflow should now be usable in both environments. 

Please give it a try and let me know if there's anything different than expected!

**Comment by cpeterso** [comment]
@bsieber-mozilla Do you want to leave this issue open to fix the resolution_map problem? Or close this issue and file a new issue for resolution_map?

**Comment by bsieber-mozilla** [comment]
@cpeterso, glad to see it working for y'all!

_closing this, [filed another](https://github.com/mozilla/jira-bugzilla-integration/issues/334) for resolution_map_
